mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/mod.rs

```diff
@@ -35,9 +37,25 @@ pub type StorageResult<T> = Result<T, StorageError>;
 /// Convenience struct for tracking read and write pointers into the file.
 #[derive(Debug, Default, Deserialize, Serialize)]
 pub(crate) struct FilePointers {
-    write: u32,
-    read_begin: u32,
-    read_end: u32,
+    write: u64,
+    read_begin: u64,
+    read_end: u64,
+}
+
+// Metadata that is saved to state file for rapidly picking up where
+// previous `RingBuffer` state was at.
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub(crate) struct RingBufferMetadata {
```

Comment:
When I was testing with large files, I found that initialization took a **long** time... I decided some sort of extra saved state would be useful for quickly getting back to a *close enough* location. Afterwards we can scan for a precise location. This way we don't have to scan large files, only a small portion.

Reply:
This happens frequently at each operation call. This is to a separate file as well. The problem I found when testing with larger files was the scanning of the file for where the pointers should be. Which could be very far off. This caused the time it took to initialized things be over a min depending on file size. Which feels really bad to me. If it is something we can be okay with, then this can be removed.

Reply:
mins is a problem, I agree. several seconds is pretty much ok. do you know why it takes so long to process a large file? reading from SSD should be 100x MB/sec

Reply:
Likely because it is more than just reading, it is the process of deserializing each block header and checking to see if it is a block closer to where the pointers should be or if we found the last block.

Reply:
I can spend some time to try and optimize the initial function after this pr and see if can get away from having to do this.. as it could very well be the case that I can try to avoid deserializing and it might be plenty fast after that.

Reply:
Okay, so I made some changes locally to test without the hint file. I ran two tests on my M1 macbook with file sizes of 0.5GiB (it_errs_on_insert_when_full and it_errs_on_insert_when_full_and_had_batched_and_removed). These tests always start with fresh files. This means it would only use the initial first loop from earlier code (scanning for a block, but not scanning for most recent). These combined took 611s (10 min!!!). So, I am okay with trying more to get this going faster, as I still suspect deserialization as a large amount of the slowness... but this is the reason why I added the metadata file hint.

Reply:
as an option, we can write metadata only on graceful shutdown. On crash - we will have to re-scan the file. Although, we still need to try to optimize the scanning later.

Reply:
Done :)